merlin/schema/schema.py

```diff
@@ -426,6 +438,120 @@ def remove_col(self, col_name: str) -> "Schema":
     def without(self, col_names: List[str]) -> "Schema":
         return self.excluding_by_name(col_names)
 
+    def with_tags(self, col_names: List[str], tags: Union[str, Tags]):
```

Comment:
Depending on what usage pattern is going to be more common. (changing more than one column with the same tags or more than one column with different tags).

We could consider combining the two args into a mapping from col name to tags `Dict[str, Union[str, Tags]]`? Same principle for `with_properties` / `with_dtypes` (the dtype one might be less natural to map to with the `is_list` `is_ragged` args?)

Reply:
My thought here was that the most common use case is probably tagging a chunk of columns as categorical or continuous, in which case they'd all get the same new tags. Not sure if that's actually the most common case though.